vowpalwabbit/feature_group.cc

```diff
@@ -30,16 +30,16 @@ void features::clear()
 void features::truncate_to(const features_value_iterator& pos)
 {
   auto i = pos._begin - values.begin();
```

Comment:
Could we make this safe? If `pos` didn't come from `values` we have UD.

Reply:
features_value_iterator is created with a ptr, so we'd likely need to enforce in its creation. In this scope I do not think we have to information to make this safe. I think this should be handled separately